frontend/src/App.vue

```diff
@@ -1,19 +1,25 @@
 <script setup lang="ts">
-import Breadcrumb from '@/components/Breadcrumb.vue';
-import Footer from '@/components/footer/Footer.vue';
-import Header from '@/components/header/Header.vue';
 import { RouterView } from 'vue-router';
+import { defineAsyncComponent, shallowRef, watch, type Component } from 'vue';
+import { useRoute } from 'vue-router';
+
+const route = useRoute();
+const layout_component = shallowRef<Component>();
+watch(
+    () => route.meta.layout ?? 'SimpleLayout',
+    (layout) => {
+        layout_component.value = defineAsyncComponent(
+            () => import(`@/layouts/${layout}.vue`)
+        );
+    },
+    { immediate: true }
+);
```

Comment:
This is interesting. Why are we dynamically import components? Is there a need for changing layout design? I thought layout is consistent maybe I am not clear for new design...

Reply:
We are going to have two layouts now, the landing page doesn't have a header and was not going to have a footer (but now it does). That's why we have this dynamic layout change. In the future, if we need a third layout, it can be easily implemented.